include/cop0.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file cop0.h
+ * @brief N64 COP0 Interface
+ * @ingroup n64sys
+ */
+
+/**
+ * @addtogroup n64sys
+ * @{
+ */
+
+#ifndef __LIBDRAGON_COP0_H
+#define __LIBDRAGON_COP0_H
+
+/** @brief Read the COP0 Count register (see also TICKS_READ). */
+#define C0_COUNT() ({ \
+    uint32_t x; \
+    asm volatile("mfc0 %0,$9":"=r"(x)); \
+    x; \
+})
+
+/** @brief Write the COP0 Count register. */
+#define C0_WRITE_COUNT(x) ({ \
+    asm volatile("mtc0 %0,$9"::"r"(x)); \
+})
+
+
+/** @brief Read the COP0 Compare register. */
+#define C0_COMPARE() ({ \
+    uint32_t x; \
+    asm volatile("mfc0 %0,$11":"=r"(x)); \
+    x; \
+})
+
+/** @brief Write the COP0 Compare register. */
+#define C0_WRITE_COMPARE(x) ({ \
+    asm volatile("mtc0 %0,$11"::"r"(x)); \
+})
+
+
+/** @brief Read the COP0 Status register. */
+#define C0_STATUS() ({ \
+    uint32_t x; \
+    asm volatile("mfc0 %0,$12":"=r"(x)); \
+    x; \
+})
+
+/** @brief Write the COP0 Status register. */
+#define C0_WRITE_STATUS(x) ({ \
+    asm volatile("mtc0 %0,$12"::"r"(x)); \
+    asm volatile("nop"); \
```

Comment:
We don't need this. If this is for cop0 hazards, it is dependent on the next instruction being executed, so we probably shouldn't have it here on this helper.

Reply:
Yes, it's for hazards. Those bugs are really hard to debug, and one additional nop in status registers manipulation seems innocuous. I think it's better to have it as a default. If there are situations where it must absolutely be absent (not sure why), I think those are the exceptions rather than the rule, and the user will know to simply call `mtc0` themselves without the macro.

Reply:
Maybe it is safer to have two then? That will protect from most of hazards (except TLB, which dragon currently does not use)